### DISCUSSION TITLE
Having issues in sending configuration data to python language server.

### DISCUSSION THREAD

**Discussion by sreetej1998** [TypeFox/monaco-languageclient#464] [Q&A]
Hi, I am trying to send the configuration from the monaco-language-client to the python language server. This is how I tried to do it via the middleware key.

```
new MonacoLanguageClient({
  name: "Your Name",
  clientOptions: {
    documentSelector: ['python'],
    middleware: {
      workspace: {
        configuration: (params, token, configuration) => {
          return [ {     pylsp: { plugins: { pylint: { enabled: true } } } } ];
        },
      },
    },
  },
  connectionProvider: {
    get: (errorHandler, closeHandler) =>
      Promise.resolve(
        createConnection(connection, errorHandler, closeHandler)
      ),
  },
})
```
The configuration was not reflected in the editor. I found that the code I wrote in the configuration callback above is not called at all. Despite being sent to the [vscode-languageserver-node](https://github.com/microsoft/vscode-languageserver-node), and   called [here](https://github.com/microsoft/vscode-languageserver-node/blob/6243e2753b9efbf6ad69d076e77af5d8a7d1f3a9/client/src/common/configuration.ts#L38-L41). 

This [Thread](https://github.com/TypeFox/monaco-languageclient/issues/201#issuecomment-648398659) might be useful on getting more context.

I am not sure If I am missing anything, would be grateful for any help, Thanks!!

**Comment by CGNonofr** [comment]
Hello,

I'm not sure how the python language server behave, but there is 2 ways to send configuration to the server:
- When the server asks for it, using the [`workspace/configuration`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_configuration) (it was introduced on the v3.6 of the protocol
- The client can push the [`workspace/didChangeConfiguration`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_didChangeConfiguration) notification to update the config on the server

The first one doesn't require any additional code from the client, while the second requires to set the `synchronize.configurationSection` on the language client

Also, it requires the configuration features to be enabled on the language client by calling `registerConfigurationFeatures` on  the language client object
